### PR TITLE
Add docs for newly added widgets

### PR DESCRIPTION
**Pull request by prateekma**
Docs for https://github.com/wpilibsuite/allwpilib/pull/2947

### CODE REVIEW

**Comment by prateekma** [comment]
Should be ready to go / review now.

File: source/docs/software/wpilib-tools/glass/command-based-widgets.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      #include <frc/smartdashboard/SmartDashboard.h>
+
+      ...
+
+      MyCommand command{...};
+      frc::SmartDashboard::PutData("My Command", &command);
+
+.. note:: The ``MyCommand`` instance can also be sent using a lower-level NetworkTables API or using the :ref:`Shuffleboard API <docs/software/wpilib-tools/shuffleboard/getting-started/shuffleboard-displaying-data:Displaying data from your robot>`. In this case, the ``SmartDashboard`` API was used, meaning that the :guilabel:`Command Selector` widget will appear under the ``SmartDashboard`` table name.
+
+.. image:: images/command-selector.png
```
Comment (TheTripleV) [inline source/docs/software/wpilib-tools/glass/command-based-widgets.rst:26]:
```suggestion
.. image:: images/command-selector.png
   :alt: Command Selector widget showing that "MyCommand" is running with the option to cancel
```

File: source/docs/software/wpilib-tools/glass/command-based-widgets.rst [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+The widget has two states. When the command is not running, a :guilabel:`Run` button will appear -- clicking it will schedule the command. When the command is running, a :guilabel:`Cancel` button, accompanied by :guilabel:`Running...` text, will appear (as shown above). This will cancel the command.
+
+Subsystem Widget
+----------------
+The :guilabel:`Subsystem` widget can be used to see the default command and the currently scheduled command on a specific subsystem. If you are using the ``SubsystemBase`` base class, your subsystem will be automatically sent to NetworkTables over LiveWindow. To view this widget, look under the :guilabel:`LiveWindow` main table name in the :guilabel:`NetworkTables` menu.
+
+.. image:: images/subsystem.png
```
Comment (TheTripleV) [inline source/docs/software/wpilib-tools/glass/command-based-widgets.rst:34]:
```suggestion
.. image:: images/subsystem.png
   :alt: Subsystem widget showing the state of "DriveSubsystem". Default Command: "DefaultDrive". Current Command: "DriveDistance"
```

File: source/docs/software/wpilib-tools/glass/command-based-widgets.rst [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+Command Scheduler Widget
+------------------------
+The :guilabel:`Command Scheduler` widget allows you to see all currently scheduled commands. In addition, any of these commands can be canceled from the GUI.
+
+.. image:: images/scheduler.png
```
Comment (TheTripleV) [inline source/docs/software/wpilib-tools/glass/command-based-widgets.rst:40]:
```suggestion
.. image:: images/scheduler.png
   :alt: Scheduler widget. Scheduled commands are "DriveDistance" and "MyCommand". Both have the option to cancel.
```

File: source/docs/software/wpilib-tools/glass/widgets.rst [hunk 4]
```diff
@@ -6,16 +6,22 @@ Specialized widgets are available for certain types that exist in robot code. Th
...
 The list of hardware (sent over LiveWindow automatically) that has widgets is below:
 
  - ``DigitalInput``
  - ``DigitalOutput``
+ - ``SpeedController``
+ - ``Gyro``
+
+Here is an example of the widget for gyroscopes:
+
+ .. image:: images/gyro.png
```
Comment (TheTripleV) [inline source/docs/software/wpilib-tools/glass/widgets.rst:24]:
```suggestion
 .. image:: images/gyro.png
    :alt: Gyro widget with both text and dial visualizations of the current gyro angle. Current Gyro Angle in degrees is -60.9704.
```

File: source/docs/software/wpilib-tools/glass/widgets.rst [hunk 5]
```diff
@@ -36,6 +42,23 @@ The :guilabel:`Sendable Chooser` widget will appear in the :guilabel:`NetworkTab
...
+
+      SmartDashboard.putData("Elevator PID Controller", m_elevatorPIDController);
+
+   .. code-tab:: c++
+
+      frc::SmartDashboard::PutData("Elevator PID Controller", &m_elevatorPIDController);
+
+This allows you to quickly tune P, I, and D values for various setpoints.
+
+.. image:: images/pid.png
```
Comment (TheTripleV) [inline source/docs/software/wpilib-tools/glass/widgets.rst:60]:
```suggestion
.. image:: images/pid.png
   :alt: PID widget for the Elevator PID Controller. P = 3.0, I = 0.001, D = 0.050, Setpoint = 0.0.
```

**Review by TheTripleV — CHANGES_REQUESTED** [review]

**Review by PeterJohnson — APPROVED** [review]

**Pull request merged**